seafile-shared-6.2.4 fails to build: http-tx-mgr.c:1115:5: error: duplicate case value

## Issue description

When trying to build seafile-share on a release-18.09 system, the build fails:

<details>
```
Making all in daemon
make[2]: Entering directory '/build/source/daemon'
gcc -DHAVE_CONFIG_H -I. -I..    -DPKGDATADIR=\"/nix/store/kwmcxvjiyyjnw49yxa04xj4h0pwzf1g5-seafile-shared-6.2.4/share/seafile\" -DPACKAGE_DATA_DIR=\""/nix/store/kwmcxvjiyyjnw49yxa04xj4h0pwzf1g5-seafile-shared-6.2.4/share/seafile"\" -DSEAFILE_CLIENT -I../include -I../lib -I../lib -I../common -I/nix/store/m5d8609dphqi3gn7dfycz75qn2i2sa1a-libsearpc-3.1.0/include -I/nix/store/m5d8609dphqi3gn7dfycz75qn2i2sa1a-libsearpc-3.1.0/include/searpc -I/nix/store/kkczf0s9g04f5sfiwzd7y0f0qj0hkrgw-glib-2.56.0-dev/include/glib-2.0 -I/nix/store/yh447r6mqpa6i047i46bplzii98l3rdy-glib-2.56.0/lib/glib-2.0/include -I/nix/store/bgijj4gfk9r6fw616dg9npi7qfg7jq64-jansson-2.11/include -I/nix/store/kkczf0s9g04f5sfiwzd7y0f0qj0hkrgw-glib-2.56.0-dev/include/glib-2.0 -I/nix/store/yh447r6mqpa6i047i46bplzii98l3rdy-glib-2.56.0/lib/glib-2.0/include -I/nix/store/v6qczmgz4ps31x7g254l9087w2nygpcf-curl-7.62.0-dev/include -Wall -g -O2 -c -o seaf-daemon.o seaf-daemon.c
seaf-daemon.c: In function 'main':
seaf-daemon.c:430:9: warning: ignoring return value of 'daemon', declared with attribute warn_unused_result [-Wunused-result]
         daemon (1, 0);
         ^~~~~~~~~~~~~
gcc -DHAVE_CONFIG_H -I. -I..    -DPKGDATADIR=\"/nix/store/kwmcxvjiyyjnw49yxa04xj4h0pwzf1g5-seafile-shared-6.2.4/share/seafile\" -DPACKAGE_DATA_DIR=\""/nix/store/kwmcxvjiyyjnw49yxa04xj4h0pwzf1g5-seafile-shared-6.2.4/share/seafile"\" -DSEAFILE_CLIENT -I../include -I../lib -I../lib -I../common -I/nix/store/m5d8609dphqi3gn7dfycz75qn2i2sa1a-libsearpc-3.1.0/include -I/nix/store/m5d8609dphqi3gn7dfycz75qn2i2sa1a-libsearpc-3.1.0/include/searpc -I/nix/store/kkczf0s9g04f5sfiwzd7y0f0qj0hkrgw-glib-2.56.0-dev/include/glib-2.0 -I/nix/store/yh447r6mqpa6i047i46bplzii98l3rdy-glib-2.56.0/lib/glib-2.0/include -I/nix/store/bgijj4gfk9r6fw616dg9npi7qfg7jq64-jansson-2.11/include -I/nix/store/kkczf0s9g04f5sfiwzd7y0f0qj0hkrgw-glib-2.56.0-dev/include/glib-2.0 -I/nix/store/yh447r6mqpa6i047i46bplzii98l3rdy-glib-2.56.0/lib/glib-2.0/include -I/nix/store/v6qczmgz4ps31x7g254l9087w2nygpcf-curl-7.62.0-dev/include -Wall -g -O2 -c -o job-mgr.o job-mgr.c
gcc -DHAVE_CONFIG_H -I. -I..    -DPKGDATADIR=\"/nix/store/kwmcxvjiyyjnw49yxa04xj4h0pwzf1g5-seafile-shared-6.2.4/share/seafile\" -DPACKAGE_DATA_DIR=\""/nix/store/kwmcxvjiyyjnw49yxa04xj4h0pwzf1g5-seafile-shared-6.2.4/share/seafile"\" -DSEAFILE_CLIENT -I../include -I../lib -I../lib -I../common -I/nix/store/m5d8609dphqi3gn7dfycz75qn2i2sa1a-libsearpc-3.1.0/include -I/nix/store/m5d8609dphqi3gn7dfycz75qn2i2sa1a-libsearpc-3.1.0/include/searpc -I/nix/store/kkczf0s9g04f5sfiwzd7y0f0qj0hkrgw-glib-2.56.0-dev/include/glib-2.0 -I/nix/store/yh447r6mqpa6i047i46bplzii98l3rdy-glib-2.56.0/lib/glib-2.0/include -I/nix/store/bgijj4gfk9r6fw616dg9npi7qfg7jq64-jansson-2.11/include -I/nix/store/kkczf0s9g04f5sfiwzd7y0f0qj0hkrgw-glib-2.56.0-dev/include/glib-2.0 -I/nix/store/yh447r6mqpa6i047i46bplzii98l3rdy-glib-2.56.0/lib/glib-2.0/include -I/nix/store/v6qczmgz4ps31x7g254l9087w2nygpcf-curl-7.62.0-dev/include -Wall -g -O2 -c -o timer.o timer.c
gcc -DHAVE_CONFIG_H -I. -I..    -DPKGDATADIR=\"/nix/store/kwmcxvjiyyjnw49yxa04xj4h0pwzf1g5-seafile-shared-6.2.4/share/seafile\" -DPACKAGE_DATA_DIR=\""/nix/store/kwmcxvjiyyjnw49yxa04xj4h0pwzf1g5-seafile-shared-6.2.4/share/seafile"\" -DSEAFILE_CLIENT -I../include -I../lib -I../lib -I../common -I/nix/store/m5d8609dphqi3gn7dfycz75qn2i2sa1a-libsearpc-3.1.0/include -I/nix/store/m5d8609dphqi3gn7dfycz75qn2i2sa1a-libsearpc-3.1.0/include/searpc -I/nix/store/kkczf0s9g04f5sfiwzd7y0f0qj0hkrgw-glib-2.56.0-dev/include/glib-2.0 -I/nix/store/yh447r6mqpa6i047i46bplzii98l3rdy-glib-2.56.0/lib/glib-2.0/include -I/nix/store/bgijj4gfk9r6fw616dg9npi7qfg7jq64-jansson-2.11/include -I/nix/store/kkczf0s9g04f5sfiwzd7y0f0qj0hkrgw-glib-2.56.0-dev/include/glib-2.0 -I/nix/store/yh447r6mqpa6i047i46bplzii98l3rdy-glib-2.56.0/lib/glib-2.0/include -I/nix/store/v6qczmgz4ps31x7g254l9087w2nygpcf-curl-7.62.0-dev/include -Wall -g -O2 -c -o cevent.o cevent.c
gcc -DHAVE_CONFIG_H -I. -I..    -DPKGDATADIR=\"/nix/store/kwmcxvjiyyjnw49yxa04xj4h0pwzf1g5-seafile-shared-6.2.4/share/seafile\" -DPACKAGE_DATA_DIR=\""/nix/store/kwmcxvjiyyjnw49yxa04xj4h0pwzf1g5-seafile-shared-6.2.4/share/seafile"\" -DSEAFILE_CLIENT -I../include -I../lib -I../lib -I../common -I/nix/store/m5d8609dphqi3gn7dfycz75qn2i2sa1a-libsearpc-3.1.0/include -I/nix/store/m5d8609dphqi3gn7dfycz75qn2i2sa1a-libsearpc-3.1.0/include/searpc -I/nix/store/kkczf0s9g04f5sfiwzd7y0f0qj0hkrgw-glib-2.56.0-dev/include/glib-2.0 -I/nix/store/yh447r6mqpa6i047i46bplzii98l3rdy-glib-2.56.0/lib/glib-2.0/include -I/nix/store/bgijj4gfk9r6fw616dg9npi7qfg7jq64-jansson-2.11/include -I/nix/store/kkczf0s9g04f5sfiwzd7y0f0qj0hkrgw-glib-2.56.0-dev/include/glib-2.0 -I/nix/store/yh447r6mqpa6i047i46bplzii98l3rdy-glib-2.56.0/lib/glib-2.0/include -I/nix/store/v6qczmgz4ps31x7g254l9087w2nygpcf-curl-7.62.0-dev/include -Wall -g -O2 -c -o http-tx-mgr.o http-tx-mgr.c
http-tx-mgr.c: In function 'curl_error_to_http_task_error':
http-tx-mgr.c:1115:5: error: duplicate case value
     case CURLE_SSL_CACERT:
     ^~~~
http-tx-mgr.c:1113:5: note: previously used here
     case CURLE_PEER_FAILED_VERIFICATION:
     ^~~~
make[2]: *** [Makefile:610: http-tx-mgr.o] Error 1
make[2]: Leaving directory '/build/source/daemon'
make[1]: *** [Makefile:443: all-recursive] Error 1
make[1]: Leaving directory '/build/source'
make: *** [Makefile:375: all] Error 2
builder for '/nix/store/wflzbibjdkbsmf59yb63pc2b4y12974s-seafile-shared-6.2.4.drv' failed with exit code 2
error: build of '/nix/store/wflzbibjdkbsmf59yb63pc2b4y12974s-seafile-shared-6.2.4.drv' failed
```
</details>


### Steps to reproduce

nix-build -E "with import <nixpkgs> {}; pkgs.seafile-shared"

## Technical details

 - system: `"x86_64-linux"`
 - host os: `Linux 4.14.82, NixOS, 18.09.1446.db561c90ef3 (Jellyfish)`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.1.3`
 - channels(root): `"nixos-18.09.1446.db561c90ef3, nixos-unstable-19.03pre159472.80738ed9dc0"`
 - channels(user): `"home-manager-18.09"`
 - nixpkgs: `/nix/var/nix/profiles/per-user/root/channels/nixos`

--- a/pkgs/applications/networking/seafile-client/default.nix
+++ b/pkgs/applications/networking/seafile-client/default.nix
@@ -5,14 +5,14 @@
 with stdenv.lib;
 
 stdenv.mkDerivation rec {
-  version = "6.2.4";
+  version = "6.2.7";
   name = "seafile-client-${version}";
 
   src = fetchFromGitHub {
     owner = "haiwen";
     repo = "seafile-client";
     rev = "v${version}";
-    sha256 = "0z5li1d7ipsm1mp0lk6mls220v4vlmsqm8ym8kxlkinqbmkky0hz";
+    sha256 = "16ikl6vkp9v16608bq2sfg48idn2p7ik3q8n6j866zxkmgdvkpsg";
   };
 
   nativeBuildInputs = [ pkgconfig cmake makeWrapper ];

--- a/pkgs/misc/seafile-shared/default.nix
+++ b/pkgs/misc/seafile-shared/default.nix
@@ -1,14 +1,14 @@
 {stdenv, fetchFromGitHub, which, autoreconfHook, pkgconfig, curl, vala, python, intltool, fuse, ccnet}:
 
 stdenv.mkDerivation rec {
-  version = "6.2.4";
+  version = "6.2.7";
   name = "seafile-shared-${version}";
 
   src = fetchFromGitHub {
     owner = "haiwen";
     repo = "seafile";
     rev = "v${version}";
-    sha256 = "0v6wvw8x3zfcapi0lk71mg89gy3gw2qmdiq6mhw6qbbrzvqprlmr";
+    sha256 = "0f8h7x6q830q4pw6f6bbykiyj3lkdlgvjzg2sdaqm4bhj2c4k1n0";
   };
 
   nativeBuildInputs = [ pkgconfig which autoreconfHook vala intltool ];

